pkg/server/commands/list_objects.go

```diff
@@ -94,9 +94,9 @@ func (q *ListObjectsQuery) evaluate(
 		if err != nil {
 			errChan <- err
 			close(errChan)
+		} else {
+			close(resultsChan)
```

Comment:
The `else` is redundant.

Instead

```go
if err != nil {
    errChan <- err
    return
}

close(resultsChan)
```

`StreamedConnectedObjects` is synchronous, but produces asynchronously on the `resultsChan`. So if we see an error, then just yield it and return.

Reply:
In your snippet you don't have the line `close(errChan)`. Is this expected? And if yes, how is it better than the way I wrote it? To be clear, from my testing, both approaches seem to work just fine.

Reply:
Sorry, I omitted the `close(errChan)` bit. That was supposed to be there. I've updated the snippet.

> And if yes, how is it better than the way I wrote it?

Less nesting == more readability. It's just a style thing mostly. In our code we see more

```go
if err != nil {
    // do something
    return
}

// do something else
```

more than we see

```go
if err != nil {
    // do something
} else {
   // do something else
}
```

And yeah, they are functionally equivalent. So it's more a stylistic pattern matching thing.